### PR TITLE
Entity type page UX improvements

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR adds a number of UX improvements to the entity type page.

## 🔗 Related links

- Redirect when namespace or entity type not found: [Asana task](https://app.asana.com/0/0/1203378271893686/f) _(internal)_
- Hide tabs before type creation: [Asana task](https://app.asana.com/0/1203301853897838/1203410573355830/f) _(internal)_
- Entities tab empty state:  [Asana task](https://app.asana.com/0/1203301853897838/1203410573355832/f) _(internal)_

## 🔍 What does this change?

- When the entity type can't be found (either because the namespace or entity type doesn't exist) the user is now redirected to the entity type creation page;
- Added a fallback in the entities tab when there are no entities to display;
- Navigation tabs in the entity type page are no longer visible before the entity type is created.

## 📹 Demo

image

image

### CODE REVIEW

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx [hunk 1]
```diff
@@ -132,6 +139,41 @@ const Page: NextPageWithLayout = () => {
...
+    if (authenticatedUser && !namespace) {
+      // eslint-disable-next-line no-console
+      console.error(
+        `Error: Couldn't find namespace with shortname '${router.query["account-slug"]}'.`,
+      );
+      void router.replace(
+        `/@${authenticatedUser.shortname}/types/new/entity-type`,
+      );
+      return;
+    }
```
Comment (nathggns) [inline packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx:160]: Curious why this is here twice?

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-entity-type.tsx [hunk 2]
```diff
@@ -63,13 +64,14 @@ export const useEntityTypeValue = (
...
         /** @todo - pick the latest version? */
-        const relevantEntityType = relevantEntityTypes
+        const relevantEntityType = relevantEntityTypes.length
           ? relevantEntityTypes[0]!.inner
           : null;
 
         await loadTypeSystem();
 
         if (!cancelled) {
+          setLoading(false);
```
Comment (nathggns) [inline packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-entity-type.tsx:74]: This also needs to be set to true when the request begins, for when `entityTypeBaseUri` changes

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx [hunk 3]
```diff
@@ -132,6 +139,41 @@ const Page: NextPageWithLayout = () => {
...
+      void router.replace(
+        `/@${authenticatedUser.shortname}/types/new/entity-type`,
+      );
+    }
+  }, [authenticatedUser, namespace, router]);
+
+  useEffect(() => {
+    if (authenticatedUser && !namespace) {
+      // eslint-disable-next-line no-console
+      console.error(
```
Comment (nathggns) [inline packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx:153]: Feels more like a warning

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx (hunk 1 above)
Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx]: Hmm I think I might have overlooked this when resolving merge conflicts. Removed the duplicated code in a1c372beb8599bda17749144e182f34a8577ea6f.

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx (hunk 3 above)
Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx]: Switched to console.warn in 5d8cf7172392b7df3f295be1c44481f3e34b8e10.

File: packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-entity-type.tsx (hunk 2 above)
Comment (luisbettencourt) [reply packages/hash/frontend/src/pages/[account-slug]/types/entity-type/use-entity-type.tsx:75]: Done in abff84d7f17d08fd0cd9136a28ce010b988d5b97.

**Review by nathggns — DISMISSED** [review]

**Review by nathggns — APPROVED** [review]

**Pull request merged**